### PR TITLE
Scroll container

### PR DESCRIPTION
**Pull request by peppy**
Only supports scrollbar based scrolling for now.

### CODE REVIEW

File: osu.Framework/Graphics/Containers/ScrollContainer.cs [hunk 1]
```diff
@@ -1,11 +1,174 @@
...
+        }
+
+        public class ScrollBar : Container
+        {
+            public Color4 HoverColour = Color4.YellowGreen;
+            public Color4 DefaultColour = Color4.White;
+            private float availableContent;
+            private float displayableContent;
+
+            public event Action<float> OnScroll;
```
Comment (Tom94) [inline osu.Framework/Graphics/Containers/ScrollContainer.cs]: I'm not sure I like that event here... having events in the update loops always makes you unsure whether the events get called after your own update method or beforehand, which can lead to some nasty bugs. Let'd discuss

File: osu.Framework/Graphics/Drawable_Interaction.cs [hunk 2]
```diff
@@ -175,7 +175,9 @@ public virtual Vector2 GetLocalPosition(Vector2 screenSpacePos)
 
         internal virtual bool Contains(Vector2 screenSpacePos)
         {
-            return ScreenSpaceInputQuad.Contains(screenSpacePos).HasValue;
+            //IMPORTANT: SizeMode Conditions added here to allow the mouse cursor to accept input outside the visible gamewindow.
+            //This works in conjunction with overridden Contains method at a parent level, and is required because of ProcessingContainer Containers.
+            return (SizeMode == InheritMode.XY && Size == Vector2.One) || ScreenSpaceInputQuad.Contains(screenSpacePos).HasValue;
```
Comment (Tom94) [inline osu.Framework/Graphics/Drawable_Interaction.cs]: I don't understand. Can you explain for which Drawables the (SizeMode == InheritMode.XY && Size == Vector2.One) gets evaluated to true and why?

Comment (peppy) [inline osu.Framework/Graphics/Drawable_Interaction.cs]: At the moment in ProcessingContainers, an extra level of container is added before any processingcontainers are added via AddProcessing to add the abstraction (and allow for easy moving of children around as more processing is potentially added).

This container does not have the overrides I added elsewhere of `Contains` so this is a temporary hack to bypass those containers.

Comment (Tom94) [inline osu.Framework/Graphics/Drawable_Interaction.cs]: Why not make an InputPassthroughContainer that overloads Contains to true, and have that be the intermediate container? Temporary hacks have the tendency to become not-so-temporary. :P

**Comment by peppy** [comment]
Something in here from `osu.Game`; don't merge until fixed.

**Pull request merged**